streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -687,12 +687,17 @@ public StreamsConfig(final Map<?, ?> props) {
 
     private Map<String, Object> getCommonConsumerConfigs() {
         final Map<String, Object> clientProvidedProps = getClientPropsWithPrefix(CONSUMER_PREFIX, ConsumerConfig.configNames());
+        final Map<String, Object> clientProvidedCustomProps = getClientCustomProps(CONSUMER_PREFIX, PRODUCER_PREFIX);
 
         checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedProps, NON_CONFIGURABLE_CONSUMER_DEFAULT_CONFIGS);
         checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedProps, NON_CONFIGURABLE_CONSUMER_EOS_CONFIGS);
 
+        checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedCustomProps, NON_CONFIGURABLE_CONSUMER_DEFAULT_CONFIGS);
+        checkIfUnexpectedUserSpecifiedConsumerConfig(clientProvidedCustomProps, NON_CONFIGURABLE_CONSUMER_EOS_CONFIGS);
```

Comment:
Why do we need this? For custom configs, we don't enforce any restrictions.

Reply:
I wanted to make sure no non-configurable properties sneaked through via the custom properties. But you are right, since the consumer/producer properties are removed form the custom properties, this is redundant.